File: newfizzbuzz-frontend/src/types/Attempt.ts
import { Quiz } from "./Quiz";
import { AttemptAnswer } from "./AttemptAnswer";

export interface Attempt {
    id: number;
    userId: number;
    quizId: number;
    quiz: Quiz;
    attemptedAt: string; // DateTime in ISO format
    correctCount: number;
    totalQuestions: number;
    timeLimit: number;
    attemptAnswers: AttemptAnswer[];
}

File: newfizzbuzz-backend/NewFizzBuzz.Api/Models/Attempt.cs
using NewFizzBuzz.Api.Models;

public class Attempt
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int QuizId { get; set; }
    public Quiz Quiz { get; set; } = null!;
    public DateTime AttemptedAt { get; set; } = DateTime.UtcNow;
    public int CorrectCount { get; set; }
    public int TotalQuestions { get; set; }
    public int TimeLimit { get; set; }

    // New relationship to AttemptAnswer
    public List<AttemptAnswer> AttemptAnswers { get; set; } = new List<AttemptAnswer>();
}
